src/__experimental__/components/input/input-presentation.spec.js

```diff
@@ -180,7 +169,19 @@ describe('InputPresentation', () => {
       assertStyleMatch({
         background: '#d9e0e4',
         borderColor: '#d9e0e4 !important'
-      }, mountRender({ theme: classicTheme, disabled: true }));
+      }, renderWithTheme({ disabled: true }, classicTheme));
     });
   });
 });
+
+function render(props, renderer = mount) {
+  return renderer(<InputPresentation { ...props }>sample children</InputPresentation>);
+}
+
+function renderWithTheme(props, theme, renderer = mount) {
```

Comment:
Yes, it's required to pass the theme to styled components. We are filtering unnecessary props, basing on PropTypes, to avoid passing `data` attributes and other html stuff to styled components. Theme is filtered out as it should be added with a provider.